types/inferschematype.d.ts

```diff
@@ -69,7 +69,7 @@ type IsPathRequired<P, TypeKey extends TypeKeyBaseType> =
   P extends { required: true | [true, string | undefined] } | ArrayConstructor | any[]
     ? true
     : P extends (Record<TypeKey, ArrayConstructor | any[]>)
-      ? P extends { default: undefined }
+      ? P extends { default: undefined } | { required: false }
```

Comment:
See my comments here: https://github.com/Automattic/mongoose/issues/12420#issuecomment-1258236649 . We shouldn't rely on `required: false`, because `required` is false by default anyway. We instead need to check if `default` is a function that returns undefined.

Reply:
👍 I will work on this